src/server/pps/server/api_server.go

```diff
@@ -1737,10 +1742,12 @@ func (a *apiServer) parentJob(
 	if err != nil {
 		return nil, err
 	}
-	if len(jobInfo.JobInfo) == 0 {
-		return nil, nil
+	for _, jobInfo := range jobInfos.JobInfo {
+		if jobInfo.PipelineVersion == pipelineInfo.Version {
+			return jobInfo.Job, nil
+		}
```

Comment:
Does it make sense to have this filtering be part of the `ListJobRequest` so that it can be done efficiently on the server?